libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs

```diff
@@ -41,5 +44,160 @@ public GarnetStatus GeoCommands<TObjectContext>(byte[] key, ref ObjectInput inpu
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destination, true, LockType.Exclusive);
+                txnManager.SaveKeyEntryToLock(key, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            try
+            {
+                var isStoreDist = false;
+                Span<ArgSlice> geoSearchParseState = stackalloc ArgSlice[input.parseState.Count + 1]; // Current MaxSize of this stactalloc will be 144 bytes ( 12 (Max Args) x 12 (sizeof ArgSlice) )
+                var currArgIdx = 0;
+                var i = 0;
+                while (i < input.parseState.Count)
+                {
+                    if (!isStoreDist && input.parseState.GetArgSliceByRef(i).ReadOnlySpan.EqualsUpperCaseSpanIgnoringCase(CmdStrings.STOREDIST))
+                    {
+                        isStoreDist = true;
+                        break;
+                    }
+                    else
+                    {
+                        geoSearchParseState[currArgIdx] = input.parseState.GetArgSliceByRef(i);
+                        currArgIdx++;
+                    }
+                    i++;
+                }
+                geoSearchParseState[currArgIdx++] = isStoreDist ? ArgSlice.FromPinnedSpan(CmdStrings.WITHDIST) : ArgSlice.FromPinnedSpan(CmdStrings.WITHHASH);
+
+                var sourceKey = key.ToArray();
+                var parseState = new SessionParseState();
+                parseState.InitializeWithArguments(geoSearchParseState.Slice(0, currArgIdx));
+
+                var searchInput = new ObjectInput(new RespInputHeader
+                {
+                    type = GarnetObjectType.SortedSet,
+                    SortedSetOp = SortedSetOperation.GEOSEARCH,
+                }, ref parseState);
+
+                SpanByteAndMemory searchOutMem = default;
+                var searchOut = new GarnetObjectStoreOutput { spanByteAndMemory = searchOutMem };
+                var status = GeoCommands(sourceKey, ref searchInput, ref searchOut, ref objectStoreLockableContext);
+                searchOutMem = searchOut.spanByteAndMemory;
+
+                if (status == GarnetStatus.WRONGTYPE)
+                {
+                    return GarnetStatus.WRONGTYPE;
+                }
+
+                if (status == GarnetStatus.NOTFOUND)
+                {
+                    _ = EXPIRE(destination, TimeSpan.Zero, out _, StoreType.Object, ExpireOption.None, ref lockableContext, ref objectStoreLockableContext);
```

Comment:
Why do we need this expire call here?

Reply:
To delete the value in the existing destination key if the source key is not there

Reply:
Thanks, a comment here would be useful.

Reply:
Done